New proposal (#4)
dafiti-group/automation-team-store-challenge: README.md

# Automation Team - Technical Challenge

---

# Greetings

Thank you to take part on this technical challenge! We designed it to be convenient, low friction and useful to you - taking into account you have given your best, and to honor that we will evaluate your job providing useful, respectful and meaningful feedback.

---

# Instructions

> You have 2 ways you can show us the best you can do: the first is the freedom for you to choose from your github projects' portfolio. The second is a challenge that we propose, if you feel more inclined to that. Choose the one which is most convenient to you, given your time availability restrictions. No matter the one you choose, the criteria we will use to evaluate will be same. After you decide which one is the best to you, answer the e-mail on which this challenge was provided to you, updating us on what option you will choose and when you will deliver the test to our evaluation (we believe it can be delivered in less then and at maximum 7 days).

Reviewer: (we believe it can be delivered in less then and at maximum 7 days)
fiquei confuso com esse then

Author: typo, vou arrumar hehehe

Author: Arrumado, tx! :smile:

## OPTION 1: YOU CHOOSE WHAT TO SHOW US

### Steps

> 1) Choose a github repository of yours that already has an existing project. This project must use a python web framework (django, flask, etc...), and provide at least a database. The more it provides, e.g. Redis, Celery tasks, etc..., the best for us to evaluate what you're capable to deliver.

Reviewer 2: ```suggestion
1) Choose an existing project from a github repository of yours. This project must use a python web framework (django, flask, etc...), and provide at least a database. The more tools it provides, e.g. Redis, Celery tasks, etc..., the better for us to evaluate what you're capable to deliver.
```

Author: Ficou ótimo @mickawaii , vou alterar aqui!

> - 2) Write a great README, explaining the problem domain you're solving with the project, its' scope and any useful "business" requirement that must be taken into account.

Reviewer: ```suggestion
 taken into account.
```
account me soa estranho aqui, assim como na linha 7, porque me remete "conta" de alguma coisa (banco, plataforma, etc...), ele tambem pode ser usado como sinonimo de "consideracao"?

Author: Justamente @rockyshimithy . "Take into account" é um termo bem comum em inglês.

3) Provide a way for us to run your application locally with all of its' requirements (python and infrastructure-wide). An alternative is to host it somewhere we can interact with the API and frontend. It would be great to document the steps required to that on the README. ;)

4) If you decide to make enhancements to the existing project, that will be awesome. E.g.: requirements updates, fixes to the test suite due to the updates, writing new tests, enhancing existing project documentation... all of that will raise even more our interest. To make it easy for us to track those enhancements, we ask you to open a new branch on the project and create a Pull Request, and to send us the link to it so that we can evaluate.

5) If you opt not to make any enhancement to the existing project, that is also valid - since you follow steps 1 to 3. ;)

### Tips

- Choose the repository wisely - the one you are most technically proud of is a good metric to show us your best work. ;)

> - We will trust you to our hearts that the work you deliver is yours, and it reflects the best of you. =D

Reviewer 2: :top:

> - You are expected to **be able to explain to us your project problem domain and its' architecture** when you go the video interview. Any code suggestions and doubts we have we will make directly into your code, in a code review format, so that we can have a practical conversation and feedback cycle. So, prepare yourself accordingly. ;)

Reviewer: mas no caso de ser um projeto que ja existe, ele vai mandar PR ainda assim? ou só o repo dele? porque se for o repo dele, como fariamos o code review?

Author: Ele vai mandar o repo dele. Quando ao "code review format", é justamente porque vamos precisar pensar em como fazer isso, já que não é exatamente um code review... Talvez usar a feature de "discussions" recente do github possa ser um caminho.
